### PR TITLE
KAFKA-10063 UnsupportedOperation when querying cleaner metrics after …

### PR DESCRIPTION
**Pull request by chia7712**
https://issues.apache.org/jira/browse/KAFKA-10063

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@hachikuji Could you take a look?

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 1]
```diff
@@ -112,9 +112,19 @@ class LogCleaner(initialConfig: CleanerConfig,
 
   private[log] val cleaners = mutable.ArrayBuffer[CleanerThread]()
 
+  /**
+   * scala 2.12 does not support maxOption so we handle the empty manually.
+   * @param f to compute the result
+   * @return the computed value or 0 if there is no cleaner
+   */
+  private def computeCleanerThread(f: Seq[CleanerThread] => Int): Int =
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: a workaround for scala 2.12

Comment (hachikuji) [reply core/src/main/scala/kafka/log/LogCleaner.scala]: Maybe we could turn this into a `foldLeft` and avoid the `map`. For example:
```scala
  private def maxOverCleanerThreads(f: CleanerThread => Int): Int = {
    cleaners.foldLeft(0, (acc: Int, thread: CleanerThread) => math.max(acc, f(thread)))
  }
```

**Review by hachikuji — COMMENTED** [review]
Thanks for the patch! Left a minor suggestion.

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the fix!

**Pull request merged**